google/auth/external_account.py

```diff
@@ -40,6 +40,7 @@
 from google.auth import credentials
 from google.auth import exceptions
 from google.auth import impersonated_credentials
+from google.auth.transport import requests
```

Comment:
I'm wondering if this breaks the intention of the transport library. The code should be agnostic to the underlying implementation.

Reply:
Oh the choose here is because I want the default headers of `google.auth.transport.requests.Request`.
For the body we are reassigning it anyway in the `TokenIntrospectionClient`.
Actually I can use a raw request and add all the headers to it if we are going to remove this dependency.

Reply:
@arithmetic1728 thoughts? Should `external_account` code pull in `requests` as a dependency?

Reply:
Oh, even with a raw `requests.Request` I'm still having `requests` as a dependency.

Reply:
Yes it's fine. `requests` is the most popular transport that most people, our client lib, and gcloud use. `urllib3` is low level and not convenient (btw `requests` wraps `urllib3` to make it easier to use). I doubt if anyone want to use `urllib3`.

Reply:
In current iteration, I already get rid of the `requests` :)